File: Livermore/Matlab/kernel_03_inner_prod.m
function kernel_03_inner_prod(LEN_1D)
    %! init
    q = 0.0;
    x = randn(1, LEN_1D);
    z = randn(1, LEN_1D);

    %! loop
    for k = 1:LEN_1D
        q = q + z(k) .* x(k);
    end

    %! array_op
    q = q + sum(z .* x);

end

File: Livermore/Python/kernel_03_inner_prod.py
def kernel_03_inner_prod(LEN_1D):
    #! init
    import numpy as np

    q = 0.0
    x = np.random.randn(LEN_1D)
    z = np.random.randn(LEN_1D)

    #! loop
    for k in range(LEN_1D):
        q = q + z[k] * x[k]

    #! array_op
    q = q + np.sum(z * x)
